src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -907,5 +908,20 @@ public static void SkipInit<T>(out T value)
             // unbox !!T
             // ret
         }
+
+
+        // Internal helper methods:
+
+        // Determines if the address is aligned at least to `alignment` bytes.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsOpportunisticallyAligned<T>(ref readonly T address, nuint alignment)
+        {
+            // `alignment` is expected to be a power of 2 in bytes.
+            // We use Unsafe.AsPointer to convert to a pointer,
+            // GC will keep alignment when moving objects (up to sizeof(void*)),
+            // otherwise alignment should be considered a hint if not pinned.
+            Debug.Assert(nuint.IsPow2(alignment));
+            return ((nuint)AsPointer(ref AsRef(in address)) & (alignment - 1)) != 0;
```

Comment:
```suggestion
            return ((nuint)AsPointer(ref AsRef(in address)) & (alignment - 1)) == 0;
```

Reply:
that was clever of me lol ;)

> NOTE: jit can optimize `x % POT_CNS == 0` to this itself

I usually write it the current way (because I think it's more logical to think about which bits should be 0), but I can change it to this if we think it's more readable.